wagtail/locks.py

```diff
@@ -132,11 +132,25 @@ def for_user(self, user):
     def get_message(self, user):
         scheduled_revision = self.object.scheduled_revision
 
-        return format_html(
-            # nosemgrep: translation-no-new-style-formatting (new-style only w/ format_html)
-            _(
-                "Page '{page_title}' is locked and has been scheduled to go live at {datetime}"
-            ),
-            page_title=self.object.get_admin_display_title(),
-            datetime=scheduled_revision.approved_go_live_at.strftime("%d %b %Y %H:%M"),
-        )
+        try:
+            return format_html(
+                # nosemgrep: translation-no-new-style-formatting (new-style only w/ format_html)
+                _(
+                    "Page '{page_title}' is locked and has been scheduled to go live at {datetime}"
+                ),
+                page_title=self.object.get_admin_display_title(),
+                datetime=scheduled_revision.approved_go_live_at.strftime(
+                    "%d %b %Y %H:%M"
+                ),
+            )
+        except AttributeError:
```

Comment:
Ah yes, I wasn't quite finished with this one. I avoided checking `isinstance(self.object, Page)` because I didn't like the idea of this module importing `wagtail.models` (which also imports this module). I'll put the import in the method then, as I guess it's unavoidable in this case...

Somewhat related, I was hoping to raise some questions in the core team meeting regarding `get_admin_display_title` and also how we should handle/avoid issues in translatable UI strings for snippets (e.g. using model's `verbose_name` while there are languages with gendered nouns). And about the possibility of just using the same strings for pages and snippets... I'll put that on the agenda for tomorrow.

Reply:
Note: please do not re-review the PR yet, there are some more tests I'd like to add. I'll request another review when it's ready.